services/rds.py

```diff
@@ -1,95 +1,290 @@
+                non_compliant_resources.append(cluster["DBClusterArn"])
+                break
+        else:
+            compliant_resources.append(cluster["DBClusterArn"])
+
     return RuleCheckResult(
-        passed=False, compliant_resources=[], non_compliant_resources=[]
+        passed=not non_compliant_resources,
+        compliant_resources=compliant_resources,
+        non_compliant_resources=non_compliant_resources,
     )
 
 
 def rds_enhanced_monitoring_enabled():
+    compliant_resources = []
+    non_compliant_resources = []
+    instances = client.describe_db_instances()["DBInstances"]
+
+    for instance in instances:
+        if instance.get("MonitoringInterval", 0) != 0:
+            compliant_resources.append(instance["DBInstanceArn"])
+        else:
+            non_compliant_resources.append(instance["DBInstanceArn"])
+
     return RuleCheckResult(
-        passed=False, compliant_resources=[], non_compliant_resources=[]
+        passed=not non_compliant_resources,
+        compliant_resources=compliant_resources,
+        non_compliant_resources=non_compliant_resources,
     )
 
 
 def rds_instance_deletion_protection_enabled():
+    compliant_resources = []
+    non_compliant_resources = []
+    instances = client.describe_db_instances()["DBInstances"]
+
+    for instance in instances:
+        if instance.get("DeletionProtection", False) != False:
+            compliant_resources.append(instance["DBInstanceArn"])
+        else:
+            non_compliant_resources.append(instance["DBInstanceArn"])
+
     return RuleCheckResult(
-        passed=False, compliant_resources=[], non_compliant_resources=[]
+        passed=not non_compliant_resources,
+        compliant_resources=compliant_resources,
+        non_compliant_resources=non_compliant_resources,
     )
 
 
 def rds_instance_public_access_check():
+    compliant_resources = []
+    non_compliant_resources = []
+    instances = client.describe_db_instances()["DBInstances"]
+
+    for instance in instances:
+        if instance.get("PubliclyAccessible") == False:
+            compliant_resources.append(instance["DBInstanceArn"])
+        else:
+            non_compliant_resources.append(instance["DBInstanceArn"])
+
     return RuleCheckResult(
-        passed=False, compliant_resources=[], non_compliant_resources=[]
+        passed=not non_compliant_resources,
+        compliant_resources=compliant_resources,
+        non_compliant_resources=non_compliant_resources,
     )
 
 
 def rds_logging_enabled():
+    compliant_resources = []
+    non_compliant_resources = []
+    clusters = client.describe_db_clusters()["DBClusters"]
+
+    for cluster in clusters:
+        if sorted(cluster["EnabledCloudwatchLogsExports"]) == ["audit", "error", "general", "slowquery"]:
+            compliant_resources.append(cluster["DBClusterArn"])
+        else:
+            non_compliant_resources.append(cluster["DBClusterArn"])
```

Comment:
RDS DB engine 별로 사용하는 log의 종류가 다르기 때문에 그에 맞는 검사가 필요해 보입니다.
별도의 dictionary로 분리하여 여러 engine을 대응할 수 있습니다.
```suggestion
    logs_for_engine = {
        "aurora-mysql": ["audit", "error", "general", "slowquery"],
        "aurora-postgresql": ["postgresql"],
        "docdb": ["audit", "profiler"]
    }

    for cluster in clusters:
        if sorted(cluster["EnabledCloudwatchLogsExports"]) == logs_for_engine.get(cluster["Engine"]):
            compliant_resources.append(cluster["DBClusterArn"])
        else:
            non_compliant_resources.append(cluster["DBClusterArn"])
```